lib/some.ml

```diff
@@ -221,6 +229,144 @@ let name_unaccent s =
   in
   copy 0 0
 
+let print_other_list conf _base list =
+  let s_title = Printf.sprintf "%s" (capitale (transl conf "see also")) in
+  Wserver.printf "<h5>%s</h5>\n" s_title;
+  Mutil.list_iter_first (fun first (fn, c) ->
+    Wserver.printf "%s<a href=\"%sm=P&v=%s&other=on\">%s</a> (%d)"
+      (if first then "" else ", ") (commd conf) (code_varenv fn) fn c)
+    list
+
+type 'a env =
+    Vlist_data of (string * (string * int) list) list
+  | Vlist_ini of string list
+  | Vlist_value of (string * (string * int) list) list
+  | Venv_keys of (string * int) list
+  | Vint of int
+  | Vstring of string
+  | Vbool of bool
+  | Vother of 'a
+  | Vnone
+
+let get_env v env = try List.assoc v env with Not_found -> Vnone
+let get_vother =
+  function
+    Vother x -> Some x
+  | _ -> None
+let set_vother x = Vother x
+let bool_val x = VVbool x
+let str_val x = VVstring x
+
+let string_to_list str =
+  let rec loop acc =
+    function
+      s ->
+        if String.length s > 0 then
+          let nbc = Name.nbc s.[0] in
+          let c = String.sub s 0 nbc in
+          let s1 = String.sub s nbc (String.length s - nbc) in
+          loop (c :: acc) s1
+        else acc
+  in loop [] str
+
+let rec eval_var conf base env xx _loc sl =
+  try eval_simple_var conf base env xx sl with
+    Not_found -> eval_compound_var conf base env xx sl
+and eval_simple_var conf base env xx =
+  function
+    [s] ->
+      begin try bool_val (eval_simple_bool_var conf base env xx s) with
+        Not_found -> str_val (eval_simple_str_var conf base env xx s)
+      end
+  | _ -> raise Not_found
+and eval_simple_bool_var _conf _base env _xx =
+  function
+  | "is_first" ->
+      begin match get_env "first" env with
+        Vbool x -> x
+      | _ -> raise Not_found
+      end
+  | _ -> raise Not_found
+and eval_simple_str_var _conf _base env _xx =
+  function
+  | "substr" -> eval_string_env "substr" env
+  | "cnt" -> eval_int_env "cnt" env
+  | "max" -> eval_int_env "max" env
+  | "tail" -> eval_string_env "tail" env
+  | "keys" ->
+      let k =
+        match get_env "keys" env with
+          Venv_keys k -> k
+        | _ -> []
+      in
+      List.fold_left
+        (fun accu (k, i) -> accu ^ k ^ "=" ^ string_of_int i ^ "&") "" k
+  | "env_key" -> eval_string_env "env_key" env
+  | "env_val" -> eval_string_env "env_val" env
+  | _ -> raise Not_found
+and eval_compound_var conf base env xx sl =
+  let rec loop =
+    function
+      [s] -> eval_simple_str_var conf base env xx s
+    | ["evar"; "p"; s] ->
```

Comment:
Non utilisé?

Reply:
d'une manière générale, je ne pense pas qu'il faille supprimer les fonctions template qui ne sont pas utilisées. On est plus souvent confronté à des situations où l'absence de certaines fonctions dans certains contexte devient gênant! Je sais que le "langage" des templates n'est pas des meilleurs, mais il existe, il y a un gros investissement dans son utilisation, et j'ai plutôt tendance à le compléter/homogénéiser!

Reply:
Si on veut ne pas supprimer ces fonctions pour pouvoir y acceder plus tard, il faut alors factoriser et créer un environnement pour les templates avec des fonctions geenerique, potentiellement utilisées par tous les templates.

Sinon, je ne suis pas pour ajouter du code qui ne sert pas, et je laisserai wikipedia donner les arguments contre ces ajouts : https://fr.wikipedia.org/wiki/YAGNI

Reply:
Je suis le premier à souhaiter une factorisation qui devrait apporter le bénéfice d'une meilleure homogénéité d'un template à l'autre. Mais ça dépasse sensiblement mes capacités! (à ce jour!!)

Reply:
Je vois. Cf mon commentaire plus bas.